Composer/packages/extensions/obiformeditor/src/Form/fields/CustomObjectArrayField.tsx

```diff
@@ -0,0 +1,231 @@
+  index: number;
+  item: { [key: string]: any };
+  itemProperties: JSONSchema6;
+  keyArray: string[];
+  hasMoveUp: boolean;
+  hasMoveDown: boolean;
+  onReorder: (a: number, b: number) => void;
+  onDelete: (idx: number) => void;
+  onEdit: (idx: number, value?: { [key: string]: any }) => void;
+}
+
+const ObjectItem: React.FC<ObjectItemProps> = props => {
+  const { item, index, itemProperties, keyArray = [], onEdit, hasMoveUp, hasMoveDown, onReorder, onDelete } = props;
+  const [key, setKey] = useState(`${item[keyArray[0]]}:${item[keyArray[1]]}`);
+
+  const contextItems: IContextualMenuItem[] = [
+    {
+      key: 'moveUp',
+      text: 'Move Up',
+      iconProps: { iconName: 'CaretSolidUp' },
+      disabled: !hasMoveUp,
+      onClick: () => onReorder(index, index - 1),
+    },
+    {
+      key: 'moveDown',
+      text: 'Move Down',
+      iconProps: { iconName: 'CaretSolidDown' },
+      disabled: !hasMoveDown,
+      onClick: () => onReorder(index, index + 1),
+    },
+    {
+      key: 'remove',
+      text: 'Remove',
+      iconProps: { iconName: 'Cancel' },
+      onClick: () => onDelete(index),
+    },
+  ];
+
+  const handleEdit = (field: string) => (_e: any, val?: string) => {
+    onEdit(index, { ...item, [field]: val });
+  };
+
+  const handleBlur = () => {
+    setKey(`${item[keyArray[0]]}:${item[keyArray[1]]}`);
+    if (!item[keyArray[0]] && !item[keyArray[1]]) {
+      onDelete(index);
+    }
+  };
+
+  return (
+    <div css={[customObjectArrayItemContainer(), customObjectArrayItem]} key={key}>
+      <div css={customObjectArrayItemValue}>
+        <EditableField
+          onChange={handleEdit(keyArray[0])}
+          value={item[keyArray[0]]}
+          placeholder={itemProperties[keyArray[0]].description}
+          styles={{
+            root: { margin: '7px 0 7px 0' },
+          }}
+          onBlur={handleBlur}
+        />
+      </div>
+      <div css={customObjectArrayItemValue}>
+        <EditableField
+          onChange={handleEdit(keyArray[1])}
+          value={item[keyArray[1]]}
+          placeholder={itemProperties[keyArray[1]].description}
+          styles={{
+            root: { margin: '7px 0 7px 0' },
+          }}
+          options={{ transparentBorder: true }}
+          onBlur={handleBlur}
+        />
+      </div>
+      <div>
+        <IconButton
+          menuProps={{ items: contextItems }}
+          menuIconProps={{ iconName: 'MoreVertical' }}
+          ariaLabel={formatMessage('Item Actions')}
+          styles={{ menuIcon: { color: NeutralColors.black, fontSize: FontSizes.size16 } }}
+        />
+      </div>
+    </div>
+  );
+};
+
+export const CustomObjectArrayField: React.FC<FieldProps> = props => {
+  const { onChange, formData = [], id, label, schema = {} } = props;
+  const description = schema.description;
+  const itemProperties = (schema.items as JSONSchema6).properties as { [k: string]: JSONSchema6 };
+  const keyArray = Object.keys(itemProperties) || [];
```

Comment:
I think `Object.keys()` always return an array.

```suggestion
  const keyArray = Object.keys(itemProperties);
```